dashboard/src/components/AppView/AppControls/StatusAwareButton.tsx

```diff
@@ -0,0 +1,32 @@
+import { CdsButton } from "@cds/react/button";
+import ReactTooltip from "react-tooltip";
+import { hapi } from "../../../shared/hapi/release";
+
+export interface IStatusAwareButtonProps {
+  id: string;
+  releaseStatus: hapi.release.IStatus | undefined | null;
+}
+
+export default function StatusAwareButton<T extends IStatusAwareButtonProps>(props: T) {
+  const { id, releaseStatus, ...otherProps } = props;
+  const disabled: boolean = releaseStatus?.code
+    ? releaseStatus.code >= 5 && releaseStatus.code <= 8
+    : false;
+  const tooltips = {
+    5: "The applicatixon is being deleted.",
+    6: "The application is pending installation.",
+    7: "The application is pending upgrade.",
+    8: "The application is pending rollback.",
+  };
+  const tooltip = releaseStatus?.code ? tooltips[releaseStatus.code] : undefined;
```

Comment:
WDYT? 

```suggestion
  const tooltip = releaseStatus?.code ? tooltips[releaseStatus.code] : "Loading application...";
```

Reply:
as stated above, the button does not handle the case of "undefined" in purpose.
and also, i believe that if the status is undefined, the whole page displays a "loading" screen before rendering the details, meaning that the buttons are not displayed

Reply:
Ok, even if we consider the `undefined` case, I agree the tooltip is not really necessary.